tests/unit/test_schemas_json_load.py

```diff
@@ -8,270 +8,763 @@
+        "type": "Invalid",
+        "given_name": "Julio",
+        "family_name": "Cesar",
+        "identifiers": [{
+            "identifier": "9999-9999-9999-9999",
+            "scheme": "Orcid"
+        }],
+        "affiliations": [{
+            "name": "Entity One",
+            "identifier": "entity-one",
+            "scheme": "entity-id-scheme"
+        }]
+    }
+    with pytest.raises(ValidationError):
+        data = CreatorSchemaV1().load(invalid_name_type)
+
+
+def test_contributor():
+    """Test contributor schema."""
+    valid_full = {
+        "name": "Julio Cesar",
+        "type": "Personal",
+        "given_name": "Julio",
+        "family_name": "Cesar",
+        "identifiers": [{
+            "identifier": "9999-9999-9999-9999",
+            "scheme": "Orcid"
+        }],
+        "affiliations": [{
+            "name": "Entity One",
+            "identifier": "entity-one",
+            "scheme": "entity-id-scheme"
+        }],
+        "role": "RightsHolder"
+    }
+
+    data = ContributorSchemaV1().load(valid_full)
+    assert data == valid_full
+
+    valid_minimal = {
+        "name": "Julio Cesar",
+        "type": "Personal",
+        "role": "RightsHolder"
+    }
+
+    data = ContributorSchemaV1().load(valid_minimal)
+    assert data == valid_minimal
+
+    invalid_no_name = {
+        "type": "Personal",
+        "given_name": "Julio",
+        "family_name": "Cesar",
+        "identifiers": [{
+            "identifier": "9999-9999-9999-9999",
+            "scheme": "Orcid"
+        }],
+        "affiliations": [{
+            "name": "Entity One",
+            "identifier": "entity-one",
+            "scheme": "entity-id-scheme"
+        }],
+        "role": "RightsHolder"
+    }
+    with pytest.raises(ValidationError):
+        data = ContributorSchemaV1().load(invalid_no_name)
+
+    invalid_no_name_type = {
+        "name": "Julio Cesar",
+        "given_name": "Julio",
+        "family_name": "Cesar",
+        "identifiers": [{
+            "identifier": "9999-9999-9999-9999",
+            "scheme": "Orcid"
+        }],
+        "affiliations": [{
+            "name": "Entity One",
+            "identifier": "entity-one",
+            "scheme": "entity-id-scheme"
+        }],
+        "role": "RightsHolder"
+    }
+    with pytest.raises(ValidationError):
+        data = ContributorSchemaV1().load(invalid_no_name_type)
+
+    invalid_name_type = {
+        "name": "Julio Cesar",
+        "type": "Invalid",
+        "given_name": "Julio",
+        "family_name": "Cesar",
+        "identifiers": [{
+            "identifier": "9999-9999-9999-9999",
+            "scheme": "Orcid"
+        }],
+        "affiliations": [{
+            "name": "Entity One",
+            "identifier": "entity-one",
+            "scheme": "entity-id-scheme"
+        }],
+        "role": "RightsHolder"
+    }
+    with pytest.raises(ValidationError):
+        data = ContributorSchemaV1().load(invalid_name_type)
```

Comment:
In case like me others are wondering (agonizing!) about what to test and the effort to value ratio, not testing the name type again is perfectly fine. We tested it in the `CreatorSchemaV1` and we know `ContributorSchemaV1` just reuses it.